inmoose/limma/contrasts.py

```diff
@@ -243,16 +243,17 @@ def contrasts_fit(fit, contrasts=None, coefficients=None):
 
     # New standard deviations
     if orthog:
-        fit.stdev_unscaled = np.sqrt(fit.stdev_unscaled**2 @ contrasts**2)
+        fit.stdev_unscaled = np.sqrt(fit.stdev_unscaled.values**2 @ contrasts**2)
     else:
         R = scipy.linalg.cholesky(cormatrix)
         ngenes = fit.stdev_unscaled.shape[0]
         ncont = contrasts.shape[1]
         U = np.ones((ngenes, ncont))
+        U = pd.DataFrame(U, index=fit.stdev_unscaled.index, columns=contrasts.columns)
         o = np.ones((1, ncoef))
         for i in range(ngenes):
-            RUC = R @ (fit.stdev_unscaled[i, :][:, None] * contrasts)
-            U[i, :] = np.sqrt(o @ RUC**2)
+            RUC = R @ (fit.stdev_unscaled.values[i, :][:, None] * contrasts)
+            U.iloc[i, :] = np.sqrt(o @ RUC**2)
```

Comment:
Not that important but I think numpy is faster than pandas, maybe you can create the DataFrame object after the for loop?